src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+					ftoa(valueFloat, tempString);
+					return;
+			}
+			sprintf(tempString, "%d", (int)valueInteger);
+		break;
+
+		case MODE_LOOKUP:
+			strcpy(tempString, lookupTables[valueTable[structureOfPage.editField[3].data].config.lookup.tableIndex].values[structureOfPage.editField[9].data]);
+        break;
+	}
+}
+
+#if (DATA_EDITION_FULL_VERSION)
+static void updatePIDValuesForThatProfile(void)
+{
+	for (int line = 3; line <= 5; line++){
+		structureOfPage.editField[firstItemOnLine(line)].data = masterConfig.profile[structureOfPage.editField[3].data].pidProfile.P8[line-3];
+		structureOfPage.editField[firstItemOnLine(line)+1].data = masterConfig.profile[structureOfPage.editField[3].data].pidProfile.I8[line-3];
+		structureOfPage.editField[firstItemOnLine(line)+2].data = masterConfig.profile[structureOfPage.editField[3].data].pidProfile.D8[line-3];
+	}
+}
+
+static void updateRateValuesForThatProfile(void)
+{
+	structureOfPage.editField[9].data = masterConfig.controlRateProfiles[structureOfPage.editField[3].data].rates[FD_ROLL];
+	structureOfPage.editField[10].data = masterConfig.controlRateProfiles[structureOfPage.editField[3].data].rates[FD_PITCH];
+	structureOfPage.editField[11].data = masterConfig.controlRateProfiles[structureOfPage.editField[3].data].rates[FD_YAW];
+	structureOfPage.editField[15].data = masterConfig.controlRateProfiles[structureOfPage.editField[3].data].dynThrPID;
+	structureOfPage.editField[18].data = masterConfig.controlRateProfiles[structureOfPage.editField[3].data].tpa_breakpoint / 10;
+}
+
+static void updateValuesForThatServo(void)
+{
+    structureOfPage.editField[6].data = currentProfile->servoConf[structureOfPage.editField[3].data].min / 10;
+    structureOfPage.editField[7].data = currentProfile->servoConf[structureOfPage.editField[3].data].max / 10;
+    structureOfPage.editField[9].data = currentProfile->servoConf[structureOfPage.editField[3].data].middle / 10;
+    structureOfPage.editField[10].data = (currentProfile->servoConf[structureOfPage.editField[3].data].rate > 0);
+    structureOfPage.editField[11].data = abs(currentProfile->servoConf[structureOfPage.editField[3].data].rate);
+    if (currentProfile->servoConf[structureOfPage.editField[3].data].forwardFromChannel == (int8_t)CHANNEL_FORWARDING_DISABLED)
+        structureOfPage.editField[12].data = 16;
+    structureOfPage.editField[18].data = servo[structureOfPage.editField[3].data] / 10;
+}
+#endif
+
+static void updateValuesForThatPage(void)
+{
+	switch (selectedPage){
+		case ALL_DATA_PAGE : {
+			updateDataValue();
+			break;
+		}
+#if (DATA_EDITION_FULL_VERSION)
+		case FEATURES_PAGE : {
+			for (int line = 0; line < bottomLineOnDisplay; line++){
+				int index = whichOfPages * DISPLAY_LINES_MINUS2 + (line);
+				structureOfPage.editField[firstItemOnLine(line+1)].data = ((featureMask() & (1 << index)) != 0);
+			}
+			break;
+		}
+        case MEASURES_PAGE : {
+        	structureOfPage.editField[3].color = !isBatteryLevelOK();
```

Comment:
color is not bool, better use `isBatteryLevelOK()?PIXEL_NORMAL:PIXEL_INVERTED`
(generated code will be identical)